pkg/controller/deliverable/reconciler.go

```diff
@@ -376,3 +381,60 @@ func getServiceAccountNameAndNamespace(deliverable *v1alpha1.Deliverable, delive
 
 	return serviceAccountName, serviceAccountNS
 }
+
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.Repo = repository.NewRepository(
+		mgr.GetClient(),
+		repository.NewCache(mgr.GetLogger().WithName("deliverable-repo-cache")),
+	)
+
+	r.ConditionManagerBuilder = conditions.NewConditionManager
+	r.ResourceRealizerBuilder = realizer.NewResourceRealizerBuilder(repository.NewRepository, realizerclient.NewClientBuilder(mgr.GetConfig()), repository.NewCache(mgr.GetLogger().WithName("deliverable-stamping-repo-cache")))
+	r.Realizer = realizer.NewRealizer()
+	r.DependencyTracker = dependency.NewDependencyTracker(
+		2*utils.DefaultResyncTime,
+		mgr.GetLogger().WithName("tracker-deliverable"),
+	)
+
+	builder := ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.Deliverable{})
+
+	m := mapper.Mapper{
+		Client:  mgr.GetClient(),
+		Logger:  mgr.GetLogger().WithName("deliverable"),
+		Tracker: r.DependencyTracker,
+	}
+
+	watches := map[client.Object]handler.MapFunc{
+		&v1alpha1.ClusterDelivery{}:  m.ClusterDeliveryToDeliverableRequests,
+		&corev1.ServiceAccount{}:     m.ServiceAccountToDeliverableRequests,
+		&rbacv1.Role{}:               m.RoleToDeliverableRequests,
+		&rbacv1.RoleBinding{}:        m.RoleBindingToDeliverableRequests,
+		&rbacv1.ClusterRole{}:        m.ClusterRoleToDeliverableRequests,
+		&rbacv1.ClusterRoleBinding{}: m.ClusterRoleBindingToDeliverableRequests,
+	}
+
+	for kindType, mapFunc := range watches {
+		// TODO: Do we need this assignment?
```

Comment:
imo even if not necessary, better do it

I imagine the builder here is intended to be used """fluently"""  and so it might be that under the hood it's indeed performing the mutations in the original object, but it could be that it's creating a new immutable on each call and never mutating the seed